File: src/leetcode/linkedlist/82-RemoveDuplicatesFromSortedListII/RemoveDuplicatesFromSortedListII.py
from typing import Optional


class ListNode:
    def __init__(self, val=0, next=None) -> None:
        self.val = val
        self.next = next

class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:

        prev, curr = head, head

        while curr:

            if prev.val == curr.val:
                prev.next = None
            else:
                prev.next = curr
                prev = curr

            curr = curr.next

        return head


s = Solution()

File: src/leetcode/linkedlist/82-RemoveDuplicatesFromSortedListII/RemoveDuplicatesFromSortedListII.rs
#[derive(PartialEq, Eq, Clone, Debug)]
pub struct ListNode {
    pub val: i32,
    pub next: Option<Box<ListNode>>,
}

impl ListNode {
    #[inline]
    fn new(val: i32) -> Self {
        ListNode { val, next: None }
    }
}

struct Solution {}

impl Solution {
    pub fn delete_duplicates(head: Option<Box<ListNode>>) -> Option<Box<ListNode>> {
        let dummy = Box::new(ListNode {
            val: -1,
            next: None,
        });
    }
}

fn main() {
    println!("going to run Solution");
}
